xds/internal/xdsclient/cds_test.go

```diff
@@ -140,6 +140,49 @@ func (s) TestValidateCluster_Failure(t *testing.T) {
 			wantUpdate: emptyUpdate,
 			wantErr:    true,
 		},
+		{
+			name: "ring-hash-hash-function-not-xx-hash",
+			cluster: &v3clusterpb.Cluster{
+				ClusterDiscoveryType: &v3clusterpb.Cluster_Type{Type: v3clusterpb.Cluster_EDS},
+				EdsClusterConfig: &v3clusterpb.Cluster_EdsClusterConfig{
+					EdsConfig: &v3corepb.ConfigSource{
+						ConfigSourceSpecifier: &v3corepb.ConfigSource_Ads{
+							Ads: &v3corepb.AggregatedConfigSource{},
+						},
+					},
+				},
```

Comment:
Based on the order in which `validateClusterAndConstructClusterUpdate` checks the fields in the `Cluster` resource, these fields are not required. I understand removing these fields will tie the tests too deeply with the implementation. But the advantage is that we have shorter and easier to read tests. Wdyt?